### PR TITLE
Feature/fix draft

### LINKED ISSUE #683: Sharing to RiotX delete the current draft
Share some text to a room with a draft.
The shared text is sent to the room (it should maybe just fill in the composer ?)
Observe that the draft is lost.

### LINKED ISSUE #952: draft gets deleted when you start typing directly after sending a reply
Steps to reproduce:

1. send a messageA
2. swipe on messageA to go into reply mode
3. (the reply preview appears)
4. type some text and tap on submit to send messageB
5. (the input box is now cleared but the reply preview above the input box is still displayed)
6. type some text to compose messageC
7. (messageB gets local echoed, the reply preview is removed and messageC is deleted from the input box)

I think it makes sense to completely clear the composer including the reply preview at step 5, so you can start typing a new message right away.

### PR DESCRIPTION
**Pull request by ganfra**
Simplify PR management to allow one. We are not listening to the db anymore to avoid waiting for transaction. 

Fixes #952
Fixes #683

### CODE REVIEW

**Comment by DoM1niC** [comment]
I have a big bug since your nice PR, if you write some Text and switch between Apps or Home, so the written Text is lost.... The Draft thing works after I write some and go back internal the App.

**Comment by ganfra** [comment]
Thanks for the remark @DoM1niC, found the cause for this issue

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/draft/DraftRepository.kt [hunk 1]
```diff
@@ -68,62 +92,15 @@ class DraftRepository @Inject constructor(@SessionDatabase private val monarchy:
...
-                        // Update the top draft
-                        Timber.d("Draft: update the top draft ${privacySafe(draft)}")
-                        topDraft.draftMode = newDraft.draftMode
-                        topDraft.content = newDraft.content
-                        topDraft.linkedEventId = newDraft.linkedEventId
-                    }
-                }
+                Timber.d("Draft: create a new draft ${privacySafe(draft)}")
+                userDraftEntity.userDrafts.clear()
+                userDraftEntity.userDrafts.add(newDraft)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/draft/DraftRepository.kt:99]: 😢

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 2]
```diff
@@ -226,52 +226,52 @@ class RoomDetailViewModel @AssistedInject constructor(
...
+            is RoomDetailAction.TapOnFailedToDecrypt -> handleTapOnFailedToDecrypt(action)
+            is RoomDetailAction.SelectStickerAttachment -> handleSelectStickerAttachment()
+            is RoomDetailAction.OpenIntegrationManager -> handleOpenIntegrationManager()
+            is RoomDetailAction.StartCall -> handleStartCall(action)
+            is RoomDetailAction.EndCall -> handleEndCall()
+            is RoomDetailAction.ManageIntegrations -> handleManageIntegrations()
+            is RoomDetailAction.AddJitsiWidget -> handleAddJitsiConference(action)
+            is RoomDetailAction.RemoveWidget -> handleDeleteWidget(action.widgetId)
+            is RoomDetailAction.EnsureNativeWidgetAllowed -> handleCheckWidgetAllowed(action)
+            is RoomDetailAction.CancelSend -> handleCancel(action)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:274]: AS sucks

**Review by bmarty — APPROVED** [review]
LGTM

**Comment by DoM1niC** [comment]
If I share text over the Share Feature and send it, so the Input field will not clean up, I know this is not totally a Draft thing but could be recommended to fix this similar to Draft.

**Pull request merged**